spec/unit/pops/types/type_calculator_spec.rb

```diff
@@ -234,6 +235,14 @@ def type_from_class(c)
       expect(calculator.infer(/^a regular expression$/).class).to eq(Puppet::Pops::Types::PRegexpType)
     end
 
+    it 'enumerator translates to PIterableType' do
+      expect(calculator.infer([1,2].each)).to be_a(Puppet::Pops::Types::PIterableType)
+    end
+
+    it 'iterable translates to PIterableType' do
+      expect(calculator.infer(Puppet::Pops::Types::Iterable.on(1))).to be_a(Puppet::Pops::Types::PIterableType)
+    end
+
```

Comment:
As noted earlier, it seems better to have two types; `Iterable` - which includes all the kinds of things for which an Iterator can be created, and `Iterator`, the thing that iterates. That way, the functions can use  a signature of `Iterable` instead of `Any`, and we can document that `Iterable` matches String, Integer, Array, Hash, and Iterator (and later possibly Range which we do not yet have as a runtime type in the Puppet Language, nor in the type system).